sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5449,9 +5449,9 @@ pi_result piextEventCreateWithNativeHandle(pi_native_handle NativeHandle,
   // Then, this ref count becomes zero when the event is completed and RT
   // calls piEventRelease().
   // However, an interop event created in this function would go through
-  // the same life cycle, meaning it will be released twice.
-  // To match with this life cycle of +2 and -2,
-  // we increase the ref count of the interop pi_event we return to SYCL RT.
+  // a different life cycle, meaning it will be released only once when RT
+  // calls piEventRelease in the sycl::event_impl destructor.
+  // So, we increase the ref count of the interop pi_event here. Refcount=1.
```

Comment:
Also, I wonder if we should really call to createEventAndAssociateQueue here as it may be having other wanted side-effects.

Reply:
Sorry, I must have been too sick.
You are right. The refcount=2 after retain().
The current code is right because we actually release this interop event twice.
Once by SYCL RT and another time when we release the queue.
That's why we needed the retain here.